### PR TITLE
[ng-jhipster] add a new method downloadFile

### PR DESCRIPTION
**Pull request by maznag**
Proposed PR for https://github.com/jhipster/generator-jhipster/issues/8234

### CODE REVIEW

**Comment by deepu105** [comment]
Also please add some tests

**Comment by deepu105** [comment]
Unit test would do

**Comment by deepu105** [comment]
once this is merged you need to make a PR to use this method in JHipster as well

**Comment by maznag** [comment]
@deepu105 can you help me please. I wanna create a spy object with a click() method for `a` like this `const spyObj = spyOn('a', 'click');` but I have this build error:
`error TS2345: Argument of type '"click"' is not assignable to parameter of type '"big" | "link" | "small" | "strike" | "sub" | "sup" | "toString" | "charAt" | "charCodeAt" | "con...'.`

File: tests/service/data-util.service.spec.ts [hunk 1]
```diff
@@ -46,5 +46,25 @@ describe('Data Utils service test', () => {
         it('should return the bytesize of the text', inject([JhiDataUtils], (service: JhiDataUtils) => {
             expect(service.byteSize('Hello Jhipster')).toBe(`10.5 bytes`);
         }));
+
+        it('should download the csv file', inject([JhiDataUtils], (service: JhiDataUtils) => {
+            // create spy object with a click() method
+            const spyObj = spyOn('a', 'click');
```
Comment (deepu105) [inline tests/service/data-util.service.spec.ts]: where do you get the `spyOn` method?  I dont see any import for it

Comment (deepu105) [reply tests/service/data-util.service.spec.ts]: also did you run the test locally first? using `npm test`

Comment (deepu105) [reply tests/service/data-util.service.spec.ts]: ok it seems like a global method. But `spyOn('a', 'click');` doesnt seem rite,  spyOn works on obejcts so should be something like `spyOn(element, 'onClick');

Comment (maznag) [reply tests/service/data-util.service.spec.ts]: spyOn is a Jest function. Sorry I think that ng-jhipster uses Jest for unit tests !

**Comment by maznag** [comment]
@deepu105 all tests have passed. Finally I use `jasmine.createSpyObj('a', ['click']);`

**Comment by deepu105** [comment]
patch version released

**Comment by deepu105** [comment]
Yes you can. Do not forget to update the lib version as well

**Comment by maznag** [comment]
Okay, thanks !

**Pull request merged**